docs/advanced_usages/scaling.md

```diff
@@ -0,0 +1,635 @@
+
+    To address this, consider increasing the hard limit for the user starting the ANTA process. You can check the current hard limit for a user by running the command `ulimit -n -H` in the terminal. To set a new limit, create the file `/etc/security/limits.d/10-anta.conf` with the following content:
+
+    ```bash
+    <user> hard nofile <value>
+    ```
+
+    Replace `<user>` with the username used to start the ANTA process, and `<value>` with the desired hard limit. The maximum value will depend on your system. After creating this file, log out of your current session and log back in for the changes to take effect.
+
+### Connection Management
+
+ANTA uses the `httpx` library as its underlying HTTP client for device connections. Each device has his own connection pool with a default maximum of `100` connections. On large fabrics, this can lead to a very high number of connections (n x 100), which may overwhelm the ANTA process. Connection pooling can be tuned via:
+
+```bash
+# Maximum number of allowable connections
+export ANTA_MAX_CONNECTIONS=100
+
+# Number of allowable keep-alive connections
+export ANTA_MAX_KEEPALIVE_CONNECTIONS=20
+
+# Time limit on idle keep-alive connections in seconds
+export ANTA_KEEPALIVE_EXPIRY=5.0
+```
+
+**Best Practice**: For large fabrics, limiting connections (5-10 per device) has shown optimal performance in testing. `ANTA_MAX_KEEPALIVE_CONNECTIONS` should be lower than `ANTA_MAX_CONNECTIONS`. See [HTTPX documentation](https://www.python-httpx.org/advanced/resource-limits/) for more details.
+
+### Timeouts Configuration
+
+ANTA provides several environment variables to control `httpx` timeouts:
+
+```bash
+# Global timeout
+export ANTA_TIMEOUT=30.0
+
+# Maximum amount of time to wait until a socket connection to the requested host is established
+export ANTA_CONNECT_TIMEOUT=30.0
+
+# Maximum duration to wait for a chunk of data to be received (for example, a chunk of the response body)
+export ANTA_READ_TIMEOUT=30.0
+
+# Maximum duration to wait for a chunk of data to be sent (for example, a chunk of the request body)
+export ANTA_WRITE_TIMEOUT=30.0
+
+# Maximum duration to wait for acquiring a connection from the connection pool
+export ANTA_POOL_TIMEOUT=30.0
+```
+
+If not set, these values default to the global `timeout` of ANTA, which is `30.0` seconds by default. The global timeout can also be set via the `--timeout` command-line option. See [HTTPX documentation](https://www.python-httpx.org/advanced/timeouts/) for more details.
```

Comment:
Same would put the text first with ANTA behavior and then explains how it can be finetuned and link to HTTPX